### PR TITLE
Removing dead code in System.Security.Cryptography.Algorithms

### PR DESCRIPTION
**Pull request by mccbraxton**
Cleaning up dead code as identified by [#17905](https://github.com/dotnet/corefx/issues/17905) in System.Security.Cryptography.Algorithms.  Included a change to the .gitignore to ignore JetBrains' .idea folder for Rider users.

### CODE REVIEW

**Comment by mccbraxton** [comment]
Closing while CI errors are addressed

**Comment by karelz** [comment]
Thanks for your contribution @mccbraxton!

BTW: You don't have to close the PR when there are CI failures - we will wait for them to pass first, before merging. Also each new commit will kick off new test run (and you can rerun manually by chatting with @dotnet-bot - but that is not typical workflow).

You can find new contributor docs (under construction) here: https://github.com/dotnet/corefx/wiki/New-contributor-Docs#contributing-guide ... there is also gitter chat room around the docs and the docs are RW for everyone if you want to help us improve them.
If you have any questions, please ask (on the PR/issue/gitter). Thanks and welcome on board!

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx [hunk 1]
```diff
@@ -166,9 +157,6 @@
   <data name="Cryptography_InvalidRsaParameters" xml:space="preserve">
     <value>The specified RSA parameters are not valid; both Exponent and Modulus are required fields.</value>
   </data>
-  <data name="Cryptography_InvalidPaddingMode" xml:space="preserve">
-    <value>Specified padding mode is not valid for this algorithm.</value>
-  </data>
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: This is used in the Unix implementation.

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx [hunk 2]
```diff
@@ -70,14 +70,11 @@
   <data name="Argument_InvalidOidValue" xml:space="preserve">
     <value>The OID value was invalid.</value>
   </data>
-  <data name="Argument_InvalidValue" xml:space="preserve">
-    <value>Value was invalid.</value>
-  </data>
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: This is used in the Unix implementation.

File: src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CngKeyLite.cs [hunk 3]
```diff
@@ -414,6 +414,8 @@ internal enum CngExportPolicies : int
         None = 0x00000000,
         AllowPlaintextExport = 0x00000002,      // NCRYPT_ALLOW_PLAINTEXT_EXPORT_FLAG
     }
+
+
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/CngKeyLite.cs]: Why was this space added?

File: src/Common/src/Interop/Windows/NCrypt/Interop.EncryptDecrypt.cs [hunk 4]
```diff
@@ -9,17 +9,17 @@ internal static partial class Interop
...
         [DllImport(Interop.Libraries.NCrypt, CharSet = CharSet.Unicode)]
         internal static extern unsafe ErrorCode NCryptEncrypt(SafeNCryptKeyHandle hKey, byte* pbInput, int cbInput, void* pPaddingInfo, byte* pbOutput, int cbOutput, out int pcbResult, AsymmetricPaddingMode dwFlags);
-
+            
         [DllImport(Interop.Libraries.NCrypt, CharSet = CharSet.Unicode)]
         internal static extern unsafe ErrorCode NCryptDecrypt(SafeNCryptKeyHandle hKey, [In] byte[] pbInput, int cbInput, void* pPaddingInfo, [Out] byte[] pbOutput, int cbOutput, out int pcbResult, AsymmetricPaddingMode dwFlags);
 
         [DllImport(Interop.Libraries.NCrypt, CharSet = CharSet.Unicode)]
         internal static extern unsafe ErrorCode NCryptDecrypt(SafeNCryptKeyHandle hKey, byte* pbInput, int cbInput, void* pPaddingInfo, byte* pbOutput, int cbOutput, out int pcbResult, AsymmetricPaddingMode dwFlags);
+
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/NCrypt/Interop.EncryptDecrypt.cs]: All changes to this file can be reverted.

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx [hunk 5]
```diff
@@ -227,7 +215,7 @@
     <value>CNG provider unexpectedly terminated encryption or decryption prematurely.</value>
   </data>
   <data name="Cryptography_Unmapped_System_Typed_Error" xml:space="preserve">
-    <value>The system cryptographic library returned error '{0}' of type '{1}'</value>
+     <value>The system cryptographic library returned error '{0}' of type '{1}'</value>
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: What was this space added?

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx [hunk 6]
```diff
@@ -86,20 +83,14 @@
     <value>Encoded OID length is too large (greater than 0x7f bytes).</value>
   </data>
   <data name="Cryptography_CSP_NoPrivateKey" xml:space="preserve">
-    <value>Object contains only the public half of a key pair. A private key must also be provided.</value>
+     <value>Object contains only the public half of a key pair. A private key must also be provided.</value>
   </data>
   <data name="Cryptography_Der_Invalid_Encoding" xml:space="preserve">
-    <value>ASN1 corrupted data.</value>
-  </data>
-  <data name="Cryptography_DSA_KeyGenNotSupported" xml:space="preserve">
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: This is used in the macOS build.

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx [hunk 7]
```diff
@@ -86,20 +83,14 @@
...
-    <value>DSA keys can be imported, but new key generation is not supported on this platform.</value>
-  </data>
+     <value>ASN1 corrupted data.</value>
+   </data>
   <data name="Cryptography_ECXmlSerializationFormatRequired" xml:space="preserve">
     <value>XML serialization of an elliptic curve key requires using an overload which specifies the XML format to be used.</value>
   </data>
-  <data name="Cryptography_ECC_NamedCurvesOnly" xml:space="preserve">
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: This is used in the macOS build.

File: src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx (hunk 5 above)
Comment (mccbraxton) [inline src/System.Security.Cryptography.Algorithms/src/Resources/Strings.resx]: Must be a leftover from a change I had to revert.  I will remove it.

**Comment by ViktorHofer** [comment]
@mccbraxton please make sure to only remove code which isn't used in any environment (Windows, Linux, OSX). In this case the CI leg warned us that you removed a resource which is in use (OSX). The problems could get even bigger as we remove code which is only executed on environments which aren't in CI yet (e.g. UWP).

File: src/Common/src/Interop/Windows/BCrypt/Cng.cs [hunk 8]
```diff
@@ -202,17 +190,17 @@ private static class BCryptGetPropertyStrings
...
-            int cbFormat = 0;
-            if (!Interop.CryptFormatObject(X509_ASN_ENCODING, 0, dwFormatStrType, IntPtr.Zero, oidValue, rawData, rawData.Length, null, ref cbFormat))
-                return null;
+           int cbFormat = 0;
+           if (!Interop.CryptFormatObject(X509_ASN_ENCODING, 0, dwFormatStrType, IntPtr.Zero, oidValue, rawData, rawData.Length, null, ref cbFormat))
+                    return null;
             StringBuilder sb = new StringBuilder((cbFormat + 1) / 2);
             if (!Interop.CryptFormatObject(X509_ASN_ENCODING, 0, dwFormatStrType, IntPtr.Zero, oidValue, rawData, rawData.Length, sb, ref cbFormat))
-                return null;
+                   return null;
```
Comment (bartonjs) [inline src/Common/src/Interop/Windows/BCrypt/Cng.cs]: This seems double-indented now?

**Comment by mccbraxton** [comment]
@ViktorHofer what's the best way to verify that values in strings.resx are in use by other environments?  I followed the documentation on the parent issue, and the recommendation was to run “msbuild /t:buildandtest” (as well as the solution-specific tests), which all passed.  I'm probably missing a crucial check to run.

**Comment by danmosemsft** [comment]
@mccbraxton just grep for the string ID inside src/System.Security.Cryptography.Algorithms/src. There are a few libraries where that won't work bcecause they construct the ID dynamically but I don't think this one does.

**Comment by stephentoub** [comment]
> just grep for the string ID inside src/System.Security.Cryptography.Algorithms/src

That may not work if files from src/Common are being pulled in and contain resource strings.  Not sure if that happens in this lib.

**Comment by mccbraxton** [comment]
I believe the failing Nano tests are related to:  [#20293](https://github.com/dotnet/corefx/pull/20293).  Would appreciate someone a little more well versed giving a second look however.

**Comment by ViktorHofer** [comment]
@dotnet-bot test this please

**Comment by ViktorHofer** [comment]
@dotnet-bot test Win x64 tests - Debug - Windows.10.Nano.Amd64.Open
@dotnet-bot test Win x64 tests - Release - Windows.10.Nano.Amd64.Open

**Comment by mccbraxton** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Release x64 Build and Test

**Comment by stephentoub** [comment]
Thanks

**Pull request merged**